File: web/src/model/victron_controller/dashboard/from_0_1_0_sensors.ts
// @ts-nocheck
import {Sensors as v0_1_0_Sensors} from './v0_1_0/Sensors'
import {Sensors as dashboard_Sensors} from './Sensors'
import {ActualF64 as dashboard_ActualF64} from './ActualF64'
import {Freshness as dashboard_Freshness} from './Freshness'
import {convert__actual_f64__from__0_1_0} from './from_0_1_0_actual-f64'

// Manual conversion: 0.1.0 Sensors -> 0.2.0 Sensors. Additive change —
// the new `session_kwh` field has no source in 0.1.0, so initialise it
// to a freshness=Unknown, value=null ActualF64. Live data overwrites
// this on the first myenergi poll. PR-session-kwh-sensor.
export function convert__sensors__from__0_1_0(from: v0_1_0_Sensors): dashboard_Sensors {
    const c = convert__actual_f64__from__0_1_0
    return new dashboard_Sensors(
        c(from.battery_soc),
        c(from.battery_soh),
        c(from.battery_installed_capacity),
        c(from.battery_dc_power),
        c(from.mppt_power_0),
        c(from.mppt_power_1),
        c(from.soltaro_power),
        c(from.power_consumption),
        c(from.grid_power),
        c(from.grid_voltage),
        c(from.grid_current),
        c(from.consumption_current),
        c(from.offgrid_power),
        c(from.offgrid_current),
        c(from.vebus_input_current),
        c(from.evcharger_ac_power),
        c(from.evcharger_ac_current),
        c(from.ess_state),
        c(from.outdoor_temperature),
        new dashboard_ActualF64(null, dashboard_Freshness.Unknown, 0),
    )
}

File: crates/dashboard-model/src/victron_controller/dashboard/from_0_1_0_sensors.rs
// Manual conversion: 0.1.0 Sensors -> 0.2.0 Sensors. Additive change —
// the new `session_kwh` field has no source in 0.1.0, so initialise it
// to a freshness=Unknown, value=None ActualF64 with since_epoch_ms=0.
// Live data overwrites this on the first myenergi poll. PR-session-kwh-sensor.

pub fn convert__sensors__from__0_1_0(
    from: &crate::victron_controller::dashboard::v0_1_0::sensors::Sensors,
) -> crate::victron_controller::dashboard::sensors::Sensors {
    use crate::victron_controller::dashboard::actual_f64::ActualF64;
    use crate::victron_controller::dashboard::freshness::Freshness;
    let convert =
        crate::victron_controller::dashboard::from_0_1_0_actual_f64::convert__actual_f64__from__0_1_0;
    crate::victron_controller::dashboard::sensors::Sensors {
        battery_soc: convert(&from.battery_soc),
        battery_soh: convert(&from.battery_soh),
        battery_installed_capacity: convert(&from.battery_installed_capacity),
        battery_dc_power: convert(&from.battery_dc_power),
        mppt_power_0: convert(&from.mppt_power_0),
        mppt_power_1: convert(&from.mppt_power_1),
        soltaro_power: convert(&from.soltaro_power),
        power_consumption: convert(&from.power_consumption),
        grid_power: convert(&from.grid_power),
        grid_voltage: convert(&from.grid_voltage),
        grid_current: convert(&from.grid_current),
        consumption_current: convert(&from.consumption_current),
        offgrid_power: convert(&from.offgrid_power),
        offgrid_current: convert(&from.offgrid_current),
        vebus_input_current: convert(&from.vebus_input_current),
        evcharger_ac_power: convert(&from.evcharger_ac_power),
        evcharger_ac_current: convert(&from.evcharger_ac_current),
        ess_state: convert(&from.ess_state),
        outdoor_temperature: convert(&from.outdoor_temperature),
        session_kwh: ActualF64 {
            value: None,
            freshness: Freshness::Unknown,
            since_epoch_ms: 0,
        },
    }
}
